src/odemis/gui/cont/tabs.py

```diff
@@ -2201,60 +2202,130 @@ def __init__(self, name, button, panel, main_frame, main_data):
         # Create new project directory on starting the GUI
         self._create_new_dir()
 
-        stage = self.tab_data_model.main.stage
-        stage_metadata = stage.getMetadata()
-        # start and end position are used for the gauge progress bar
-        self._start_pos = stage.position.value
-        self._end_pos = self._start_pos
-        # Show current position of the stage via the progress bar
-        main_data.stage.position.subscribe(self._update_progress_bar, init=False)
-        try:
-            self.ion_to_sample = stage_metadata[model.MD_ION_BEAM_TO_SAMPLE_ANGLE]
-        except KeyError:
-            raise ValueError('The stage is missing an ION_BEAM_TO_SAMPLE_ANGLE metadata.')
-        # Define axis connector to link milling angle to UI float ctrl
-        self.milling_connector = AxisConnector('rx', stage, panel.ctrl_milling, pos_2_ctrl=self._milling_angle_changed,
-                                               ctrl_2_pos=self._milling_ctrl_changed, events=wx.EVT_COMMAND_ENTER)
-        # Set the milling angle range according to rx axis range
-        try:
-            rx_range = stage.axes['rx'].range
-            milling_range = [-(self.ion_to_sample - a) for a in rx_range]
-            # sort milling range in case ion_to_sample made range values flip
-            actual_rng = sorted(milling_range)
-            ctrl_rng = max(actual_rng[0], MILLING_ANGLE_RANGE[0]), min(actual_rng[1], MILLING_ANGLE_RANGE[1])
-            if not ctrl_rng[0] <= DEFAULT_MILLING_ANGLE <= ctrl_rng[1]:
-                raise ValueError("Default milling angle %s should be within calculated milling range %s" % (DEFAULT_MILLING_ANGLE, ctrl_rng))
-            panel.ctrl_milling.SetValueRange(*(math.degrees(r) for r in ctrl_rng))
-            # Default value for milling angle, will be used to store the angle value out of milling position
-            self._prev_milling_angle = DEFAULT_MILLING_ANGLE
-            self.panel.ctrl_milling.Value = readable_str(math.degrees(DEFAULT_MILLING_ANGLE), unit="°", sig=3)
-        except KeyError:
-            raise ValueError('The stage is missing an rx axis.')
-        panel.ctrl_milling.Bind(wx.EVT_CHAR, panel.ctrl_milling.on_char)
-
-        self.position_btns = {LOADING: self.panel.btn_switch_loading, IMAGING: self.panel.btn_switch_imaging,
+        if main_data.role == 'enzel':  
+            try:
+                # start and end position are used for the gauge progress bar
+                self._start_pos = main_data.stage.position.value
+                self._end_pos = self._start_pos
+                # Show current position of the stage via the progress bar
+                main_data.stage.position.subscribe(self._update_progress_bar, init=False)
+                # get the stage and its meta data 
+                stage = self.tab_data_model.main.stage
```

Comment:
I think it's better to have the stage as attribute of the tab `self.stage` and point it to either the main stage in case of `enzel` or the underlying stage in case of `meteor`. After that you can use self.stage in subsequent code without having to check which microscope.

Reply:
You could also save the `self.tab_data_model.main.stage` as `self.role` as you're using it alot along the tab functions.